web/src/redux/apis/TraceTest.api.ts

```diff
@@ -1,240 +1,27 @@
 import {createApi, fetchBaseQuery} from '@reduxjs/toolkit/query/react';
-import {HTTP_METHOD} from 'constants/Common.constants';
-import {PaginationResponse} from 'hooks/usePagination';
-import {uniq} from 'lodash';
-import Environment from 'models/__mocks__/Environment.mock';
-import KeyValueMock from 'models/__mocks__/KeyValue.mock';
-import TransactionMock from 'models/__mocks__/Transaction.mock';
-import AssertionResults from 'models/AssertionResults.model';
-import Test from 'models/Test.model';
-import TestRun from 'models/TestRun.model';
-import {IEnvironment} from 'pages/Environments/IEnvironment';
-import WebSocketService, {IListenerFunction} from 'services/WebSocket.service';
-import {TAssertion, TAssertionResults, TRawAssertionResults} from 'types/Assertion.types';
-import {TRawTest, TTest} from 'types/Test.types';
-import {TRawTestRun, TTestRun} from 'types/TestRun.types';
-import {TRawTestSpecs} from 'types/TestSpecs.types';
-import {TTransaction} from 'types/Transaction.types';
-import {IKeyValue, SortBy, SortDirection} from '../../constants/Test.constants';
+import {TracetestApiTags} from 'constants/Test.constants';
+import {TTestApiEndpointBuilder} from '../../types/Test.types';
+import EnvironmentEndpoint from './endpoints/Environment.endpoint';
+import TestEndpoint from './endpoints/Test.endpoint';
+import TestRunEndpoint from './endpoints/TestRun.endpoints';
+import TransactionEndpoint from './endpoints/Transaction.endpoint';
 
 const PATH = `${document.baseURI}api/`;
 
-enum Tags {
-  ENVIRONMENT = 'environment',
-  TRANSACTION = 'transaction',
-  TEST = 'test',
-  TEST_DEFINITION = 'testDefinition',
-  TEST_RUN = 'testRun',
-  SPAN = 'span',
-}
-
-function getTotalCountFromHeaders(meta: any) {
-  return Number(meta?.response?.headers.get('x-total-count') || 0);
-}
-
 const TraceTestAPI = createApi({
   reducerPath: 'tests',
   baseQuery: fetchBaseQuery({
     baseUrl: PATH,
   }),
-  tagTypes: Object.values(Tags),
```

Comment:
This was getting out of hand so I refactored to spread the endpoints to different files